misk/src/main/kotlin/misk/web/HttpActionScopeSeedDataInterceptor.kt

```diff
@@ -0,0 +1,22 @@
+package misk.web
+
+import com.google.inject.Key
+import misk.Action
+import misk.scope.ActionScope
+
+/** Allows users to seed the [ActionScope] entered when handling an HTTP request. */
+interface HttpActionScopeSeedDataInterceptor {
+  /**
+   * Accepts the current seed data, along with the [PathPattern] and [Action] for the [BoundAction]
+   * that is being handled.
+   *
+   * Returns a new seed data map which is used to enter the [ActionScope].
+   */
+  fun intercept(
```

Comment:
I’m just pondering alternative approaches.

We’ve got this:

```
class ActionScope {
  fun enter(seedData: Map<Key<*>, Any?>): ActionScope { ... }
  fun close() { ... }
}
```

We could add something like this?

```
interface ActionScopeInterceptor {
  fun enter(seedData: Map<Key<*>, Any?>, chain: Chain): AutoCloseable

  interface Chain {
    fun proceed(seedData: Map<Key<*>, Any?>): AutoCloseable
  }
}
```

The upside there is I could seed closeable things, or do other work when the scope is closed.

You’ll also see this doesn’t declare web stuff . . . I’ve omitted the path pattern & action. I think if you need these, we do ’em in a factory for the interceptors. Again so we can do eager validation and optimize things.